django/thunderstore/community/api/experimental/views/listing.py

```diff
@@ -110,19 +101,11 @@ def post(self, request, *args, **kwargs):
 
         request_serializer = PackageListingApproveRequestSerializer(data=request.data)
         request_serializer.is_valid(raise_exception=True)
-        params = request_serializer.validated_data
-
-        try:
-            listing.approve(
-                agent=request.user,
-                internal_notes=params.get("internal_notes"),
-            )
-            listing.clear_review_request()
-            get_package_listing_or_404.clear_cache_with_args(
-                namespace=listing.package.namespace.name,
-                name=listing.package.name,
-                community=listing.community,
-            )
-            return Response(status=status.HTTP_200_OK)
-        except PermissionError:
-            raise PermissionDenied()
+
+        approve_package_listing(
+            notes=request_serializer.validated_data.get("internal_notes"),
+            user=request.user,
+            listing=listing,
+        )
```

Comment:
translating the `PermissionError` into `PermissionDenied()` as the previous implementation did still would make sense here, but as noted in the PR review head comment it can also be done on a project-wide systemic level if you don't want to deal with view specific handling like that.